New plugin minhateca.com.br box

techdealer opened the pull request.

Th3-822 commented:
Why not use web upload?
It's easier and it does provide the download link after upload, the methods are on the download plugin.

techdealer commented:
Web upload usually fail with large files (I think > 1GB). It's a limitation from the site.
My implementation works with larger files.

Both methods have their advantages:
Web upload: seems to be faster.
Box: allow upload larger files.

Th3-822 commented:
It seems that you can use the /IChomikBoxService/Download request to get the metadata of the file, so you can use it to build the download link. (globalId / name(without ext),ID,ext(from name))

Have you tested uploading without a custom upfile function? (as custom upload functions needs more maintenance and may break on updates)

1: Build the download link after upload with the Download request data.
2: Check if it can upload fine by using http.php's upfile (it should)
3: Remove the "Connection: Keep-Alive" header as only cURL() supports it.

techdealer commented:
1: I implemented the download link. Thanks for the tip.

2: I did some tests before, but this does not seem to be possible. This upload method require a custom bound, some parameters in postdata with a different type and use a different Content-Type.

The only way was rewrite http.php's upfile function.

3: Removed :)

Th3-822 commented:
btw, i did a successful upload with: `$upfiles = upfile($upload_host, $upload_port, '/file/', 0, 0, ['chomik_id' => $chomik_id, 'folder_id' => 0, 'key' => $upload_key, 'time' => $upload_time, 'client' => 'MinhaBox.br-2.0.8.1', 'locale' => 'BR'], $lfile, $lname, 'file', '', $_GET["uproxy"], $pauth, 0);`

techdealer commented:
I think everything should be ok now. :)
I was uploading a large file, and it worked.

Th3-822 commented:
Looks fine now, merged.

techdealer commented:
The upload does not fail, but found that all the files I upload are corrupted.
I think that the problem is from http.php's upfile function. With my custom function this did not happen.

Th3-822 commented:
I forgot to test the download, send another pull and i will merge it as soon as i can
